prow/plugins/cherrypickunapproved/cherrypick-unapproved_test.go

```diff
@@ -88,80 +90,113 @@ type fakePruner struct{}
-			expectComment: false,
+			name:           "has cpUnapproved -> no-op",
+			branch:         "release-1.10",
+			previousBranch: "",
+			action:         github.PullRequestActionOpened,
+			labels:         []string{labels.CpUnapproved},
+			added:          []string{},
+			removed:        []string{},
+			expectComment:  false,
 		},
 		{
-			name:          "has both cpApproved and cpUnapproved -> remove cpUnapproved",
-			branch:        "release-1.10",
-			action:        github.PullRequestActionOpened,
-			labels:        []string{labels.CpApproved, labels.CpUnapproved},
-			added:         []string{},
-			removed:       []string{labels.CpUnapproved},
-			expectComment: false,
+			name:           "has both cpApproved and cpUnapproved -> remove cpUnapproved",
+			branch:         "release-1.10",
+			previousBranch: "",
+			action:         github.PullRequestActionOpened,
+			labels:         []string{labels.CpApproved, labels.CpUnapproved},
+			added:          []string{},
+			removed:        []string{labels.CpUnapproved},
+			expectComment:  false,
 		},
 		{
-			name:          "does not have any labels, PR opened against a release branch -> add cpUnapproved and comment",
-			branch:        "release-1.10",
-			action:        github.PullRequestActionOpened,
-			labels:        []string{},
-			added:         []string{labels.CpUnapproved},
-			removed:       []string{},
-			expectComment: true,
+			name:           "does not have any labels, PR opened against a release branch -> add cpUnapproved and comment",
+			branch:         "release-1.10",
+			previousBranch: "",
+			action:         github.PullRequestActionOpened,
+			labels:         []string{},
+			added:          []string{labels.CpUnapproved},
+			removed:        []string{},
+			expectComment:  true,
 		},
 		{
-			name:          "does not have any labels, PR reopened against a release branch -> add cpUnapproved and comment",
-			branch:        "release-1.10",
-			action:        github.PullRequestActionReopened,
-			labels:        []string{},
-			added:         []string{labels.CpUnapproved},
-			removed:       []string{},
-			expectComment: true,
+			name:           "does not have any labels, PR reopened against a release branch -> add cpUnapproved and comment",
+			branch:         "release-1.10",
+			previousBranch: "",
+			action:         github.PullRequestActionReopened,
+			labels:         []string{},
+			added:          []string{labels.CpUnapproved},
+			removed:        []string{},
+			expectComment:  true,
+		},
+		{
+			name:           "PR base branch master edited to release -> add cpUnapproved and comment",
+			branch:         "release-1.10",
+			previousBranch: "master",
+			action:         github.PullRequestActionEdited,
+			labels:         []string{},
+			added:          []string{labels.CpUnapproved},
+			removed:        []string{},
+			expectComment:  true,
+		},
+		{
+			name:           "PR base branch edited from release to master -> remove cpApproved and cpUnapproved",
```

Comment:
Nit: A testcase for graceful handling of edit event that is not a branch change?